templates/ferm_chain_header.conf.epp

```diff
@@ -1,12 +1,14 @@
-<%- | Ferm::Policies $policy,
+<%- | Optional[Ferm::Policies] $policy,
       Boolean $disable_conntrack,
 | -%>
 # THIS FILE IS MANAGED BY PUPPET
+<%- if $policy { -%>
 # Default policy for this chain
 policy <%= $policy %>;
+<%- } -%>
 
 <% unless $disable_conntrack { -%>
 # connection tracking
-mod state state INVALID DROP;
-mod state state (ESTABLISHED RELATED) ACCEPT;
+mod conntrack ctstate (ESTABLISHED RELATED) ACCEPT;
```

Comment:
do those statements create different iptables rules?

Reply:
Yes they do, but both the `state` and the `ctstate` modules are bundled with iptables since forever, the change should even work on systems as old as CentOS 5 without issues.
Also the semantics of the generated rules does not change, they still do the same thing, the `ctstate` module simply is the successor of the `state` module and more efficient in its implementation and should always be favored over the legacy `state` module.